docs/db.md

```diff
@@ -1,4 +1,200 @@
+
+## Running a pool of database connections
+
+Now let's focus on the Server!
+
+All you need to do is open one time the database, and share the connection accross all your services:
+
+```haskell
+{-# language FlexibleContexts      #-}
+{-# language OverloadedStrings     #-}
+{-# language PartialTypeSignatures #-}
+{-# language TypeApplications      #-}
+{-# language TypeFamilies          #-}
+
+module Server where
+
+import           Control.Monad.Logger
+import           Mu.GRpc.Server
+import           Mu.Server
+
+main :: IO ()
+main =
+  runStderrLoggingT $
+    withSqliteConn @(LoggingT IO) "example.db" $ \conn ->
+      liftIO $ runGRpcApp 8080 (server conn)
+```
+
+We have decided in this example to use `LoggintT` from `monad-logger` and `runStderrLoggingT` to get some basic database logs to the console for free, but this is not a must!
+
+## This actually does not work
+
+Maybe you might have noticed that this example is not going to work yet, until we tell `persistent` what is our migration. We need a small tweak in our `Schema.hs`:
+
+```diff
+- mkPersist sqlSettings [persistLowerCase|
++ share [mkPersist sqlSettings, mkMigrate "migrateAll"] [persistLowerCase|
+  Person json
+  ...
+```
+
+And another one on our `Server.hs`:
+
+```diff
+main :: IO ()
+main =
+  runStderrLoggingT $
+-   withSqliteConn @(LoggingT IO) "example.db" $ \conn ->
++   withSqliteConn @(LoggingT IO) "example.db" $ \conn -> do
++     liftIO $ flip runSqlPersistM conn $ runMigration migrateAll
+      liftIO $ runGRpcApp 8080 (server conn)
+```
+
+## Sample usage with a service
+
+All the pieces are now in place, let's check the implementation of the `allPeople` service:
+
+```haskell
+allPeople :: SqlBackend
+          -> ConduitT (Entity Person) Void ServerErrorIO ()
+          -> ServerErrorIO ()
+allPeople conn sink = runDb conn $
+  runConduit $ selectSource [] [] .| liftServerConduit sink
+```
+
+As you can see, all the services need to be passed the `SqlBackend` connection as an argument.
+
+Two interesting things we want to highlight here: we have provided a small helper called `runDb`, it's implementation is quite simple and it exist due to **developer ergonomics**. We are basically saving you from writing lots of `liftIO $ flip runSqlPersistM`. 😉
+
+The second one will be discussed in the next section.
+
+## On streams and `Conuit`
+
+Since we are going to work with streams, it is wonderful that `persistent` also provides methods to work with `Conduit` like, for example, `selectSource`. But the Monad in which `persistent` operates returns `ConduitM () (Entity Person) m ()`, and we know that we want instead: `ConuitT (Entity Person) Void ServerErrorIO ()`. 🤔
+
+Well, have no fear my friend because we created yet another utility called `liftServerConduit`. Born specifically to address this problem, it's type signature is:
```

Comment:
What is this type signature telling us. I would add something below in the lines of:

> That is, we can turn any of the conduits given as input, which work on the `ServerErrorIO` monad from `mu-rpc`, into a conduit working on other `IO`-like monad. This is the case, in particular, of the monad in which Persistent runs.